test/core/utils/matches.js

```diff
@@ -132,38 +132,87 @@ describe('utils.matches', function() {
+        var virtualNode = queryFixture('<h1 id="target">foo</h1>');
+        assert.isTrue(matches(virtualNode, 'h1:not([class])'));
+      });
 
-    it('returns false if :not matches element', function() {
-      var virtualNode = queryFixture('<h1 id="target">foo</h1>');
-      assert.isFalse(matches(virtualNode, 'h1:not([id])'));
+      it('returns true if :not matches using id', function() {
+        var virtualNode = queryFixture('<h1 id="target">foo</h1>');
+        assert.isTrue(matches(virtualNode, 'h1:not(#foo)'));
+      });
+
+      it('returns true if :not matches none of the selectors', function() {
+        var virtualNode = queryFixture('<h1 id="target">foo</h1>');
+        assert.isTrue(matches(virtualNode, 'h1:not([role=heading], span)'));
+      });
+
+      it('returns false if :not matches element', function() {
+        var virtualNode = queryFixture('<h1 id="target">foo</h1>');
+        assert.isFalse(matches(virtualNode, 'h1:not([id])'));
+      });
+
+      it('returns false if :not matches one of the selectors', function() {
+        var virtualNode = queryFixture('<h1 id="target">foo</h1>');
+        assert.isFalse(matches(virtualNode, 'h1:not([role=heading], h1)'));
+      });
     });
 
-    it('throws error if pseudo is not implemented', function() {
-      var virtualNode = queryFixture('<h1 id="target">foo</h1>');
-      assert.throws(function() {
-        matches(virtualNode, 'h1:empty');
+    describe(':is', function() {
+      it('returns true if :is matches using tag', function() {
+        var virtualNode = queryFixture('<h1 id="target">foo</h1>');
+        assert.isTrue(matches(virtualNode, ':is(h1)'));
       });
-      assert.throws(function() {
-        matches(virtualNode, 'h1::before');
+
+      it('returns true if :is matches using class', function() {
+        var virtualNode = queryFixture('<h1 id="target" class="foo">foo</h1>');
+        assert.isTrue(matches(virtualNode, 'h1:is(.foo)'));
+      });
+
+      it('returns true if :is matches using attribute', function() {
+        var virtualNode = queryFixture('<h1 id="target" class="foo">foo</h1>');
+        assert.isTrue(matches(virtualNode, 'h1:is([class])'));
+      });
+
+      it('returns true if :is matches using id', function() {
+        var virtualNode = queryFixture('<h1 id="target">foo</h1>');
+        assert.isTrue(matches(virtualNode, 'h1:is(#target)'));
+      });
+
+      it('returns true if :is matches one of the selectors', function() {
+        var virtualNode = queryFixture('<h1 id="target">foo</h1>');
+        assert.isTrue(matches(virtualNode, ':is([role=heading], h1)'));
+      });
+
+      it('returns true if :is matches complex selector', function() {
+        var virtualNode = queryFixture('<div><h1 id="target">foo</h1></div>');
+        assert.isTrue(matches(virtualNode, 'h1:is(div > #target)'));
+      });
+
+      it('returns false if :is does not match element', function() {
```

Comment:
Can you add a test showing it can return false for multiple selectors, where all are false?